gradle/tasks/docker.gradle

```diff
@@ -0,0 +1,58 @@
+buildscript {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+    }
+    dependencies {
+        classpath "com.bmuschko:gradle-docker-plugin:${gradleDockerPluginVersion}"
+    }
+}
+
+apply plugin: com.bmuschko.gradle.docker.DockerRemoteApiPlugin
+
+import com.bmuschko.gradle.docker.tasks.image.DockerBuildImage
+
+task dockerBuildImageWeb(type: DockerBuildImage) {
+    group 'Docker'
+    description 'Builds the basic web server-only Docker image'
+    // Change to 'dependsOn' when we get our up-to-date sorted
+    mustRunAfter allprojects.collect { it.tasks.matching { it.name.equals('tomcatDeploy') } }
+
+    dockerFile file('docker/Dockerfile')
+    inputDir rootProject.projectDir
+    tag 'apereo/uportal'
+
+}
+
+task dockerBuildImageCli(type: DockerBuildImage) {
+    group 'Docker'
+    description 'Builds the Docker image used to invoke Import/Export tasks from within a container'
+    // Change to 'dependsOn' when we get our up-to-date sorted
+    mustRunAfter allprojects.collect { it.tasks.matching { it.name.equals('tomcatDeploy') } }
```

Comment:
:question: `mustRunAfter` or `dependsOn`?

Reply:
I'd like to use `dependsOn`, but if we set it up that way now all the `tomcatDeploy` (and ancillary) tasks will always run whenever you run any of the `dockerBuildImage...` tasks because we haven't implemented up-to-date yet.  Let's switch when we have that bit sorted.